### PR TITLE
TST: Fix devdeps job

### PR DESCRIPTION
**Pull request by pllim**
We are letting real bugs through waiting on shapely to acknowledge numpy 2.0, so I think it is less risky to disable numpy-dev for now so we can test dev versions of the rest.

<hr/>

TST: Add pyerfa-dev to devdeps and remove unnecessary tox requirements.

BLD: No longer need to specify wheel in build requirements.

xref https://github.com/astropy/astropy/issues/15638

### Blocked by

* https://github.com/scikit-image/scikit-image/issues/7249
* https://github.com/glue-viz/glue/issues/2469
* https://github.com/shapely/shapely/issues/1949

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [x] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by pllim** [comment]
Tsk tsk, need to wait for scikit-image. Same problem as https://github.com/spacetelescope/acstools/pull/192

**Comment by pllim** [comment]
Gah, now it is shapely. Do we really need this, @astrofrog ?

```
glue/core/data_factories/helpers.py:30: in <module>
    from glue.core.component import CoordinateComponent
glue/core/component.py:5: in <module>
    import shapely
shapely/__init__.py:1: in <module>
    from shapely.lib import GEOSException  # NOQA
E   ImportError: numpy.core.multiarray failed to import
```

**Comment by pllim** [comment]
Let's disable numpy-dev for now. When this is merged, I can open follow-up issue. Did fix a real test bug. I think this is ready for review/merge. Thanks!

File: jdaviz/configs/imviz/tests/test_catalogs.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def test_plugin_image_with_result(self, imviz_helper, tmp_path):
         self.imviz = imviz_helper
 
         catalogs_plugin = self.imviz.app.get_tray_item_from_name('imviz-catalogs')
-        catalogs_plugin._obj.plugin_opened = True
+        catalogs_plugin.plugin_opened = True
```
Comment (kecnry) [inline jdaviz/configs/imviz/tests/test_catalogs.py:94]: why this change?  `plugin_opened` isn't exposed through the user API afaik.  If you want to use public API, use `with plugin.as_active():` instead.

Comment (pllim) [reply jdaviz/configs/imviz/tests/test_catalogs.py:94]: It was failing the test. I removed the `_obj` to be consistent with the rest of this test module. The way this plugin object is accessed, I don't think it goes through the public API? I think these tests were written before public API concept was a thing.

Comment (kecnry) [reply jdaviz/configs/imviz/tests/test_catalogs.py:94]: ah, it uses `get_tray_item_from_name`, ok, nevermind then!

**Comment by pllim** [comment]
@kecnry , so did you approve?

**Review by kecnry — APPROVED** [review]
ok by me as long as we re-introduce testing numpy dev once upstream blockers are removed.  Thanks!

**Comment by pllim** [comment]
Opened follow-up issue at https://github.com/spacetelescope/jdaviz/issues/2659

**Comment by pllim** [comment]
@meeseeksdev backport to v3.8.x
